test/suites/test_crud.py

```diff
@@ -0,0 +1,584 @@
+                    ],
+            },
+        },
+        'crud_count': {
+            'correct_input': {
+                'args': ['tester', [['==', 'bucket_id', 100]], {'timeout': 10}],
+            },
+            'correct_output': {
+                'scalar': 9,
+            },
+            'exception_input': {
+                'args': ['no-such-space-name', [['==', 'bucket_id', 100]], {'timeout': 10}],
+            },
+            'exception_output': {
+                'str': [
+                        r'CountError: Space "no-such-space-name" doesn\'t exist', 
+                    ],
+            },
+        },
+        'crud_unflatten_rows': {
+            'correct_input': {
+                'args': [
+                    [
+                        [1, 100, 'Mike'], 
+                        [2, 100, 'Mike'], 
+                        [3, 100, 'Mike'], 
+                        [4, 100, 'Mike'], 
+                        [5, 200, 'Bill'], 
+                        [6, 300, 'Rob'],
+                    ],
+                    [
+                        {'name': 'id', 'type': 'unsigned'}, 
+                        {'name': 'bucket_id', 'type': 'unsigned'}, 
+                        {'name': 'name', 'type': 'string'}
+                    ],
+                ],
+            },
+            'correct_output': {
+                'scalar': [
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 1}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 2}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 3}, 
+                    {'bucket_id': 100, 'name': 'Mike', 'id': 4}, 
+                    {'bucket_id': 200, 'name': 'Bill', 'id': 5}, 
+                    {'bucket_id': 300, 'name': 'Rob', 'id': 6},
+                ],
+            },
+            'exception_input': {
+                'args': [[],[]],
+            },
+            'exception_output': {
+                'str': [],
+            },
+        },
+    }
+
+    def test_crud_module_via_connection(self):
+        for case_name in self.crud_test_cases.keys():
+            with self.subTest(name=case_name):
+                case = self.crud_test_cases[case_name]
+                testing_function = getattr(self.conn, case_name)
+                # Correct try.
+                try:
+                    resp = testing_function(
+                        *case['correct_input']['args'],
+                    )
+                    if 'rows' in case['correct_output']:
+                        self.assertEqual(resp.rows, case['correct_output']['rows'])
+                    if 'scalar' in case['correct_output']:
+                        # Case for scalar value as crud responce, not tarantool.crud.CrudResult obj.
+                        self.assertEqual(resp, case['correct_output']['scalar'])
+                except CrudModuleError as e:
+                    self.assertEqual(e, None)
+                # Exception try.
```

Comment:
Since it is already the two different code blocks which test the different behavior, let's make two separate dicts with cases and two separate test methods.

Reply:
Done